app/src/main/java/org/standardnotes/notes/NoteActivity.kt

```diff
@@ -1,24 +1,98 @@
 package org.standardnotes.notes
 
+import android.animation.Animator
 import android.content.Intent
 import android.os.Bundle
 import android.support.v4.app.NavUtils
 import android.view.MenuItem
+import android.view.View
+import android.view.ViewAnimationUtils
+import android.view.ViewTreeObserver
 import org.standardnotes.notes.frag.NoteFragment
 
+
 class NoteActivity : BaseActivity() {
 
+    val REVEAL_ANIM_DURATION = 200L
+    var backPressed = false
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        overridePendingTransition(0, 0)
 
-        //title = note?.title ?: "New note"
         if (savedInstanceState == null) {
             val frag: NoteFragment = NoteFragment()
             frag.arguments = intent.extras
+
             supportFragmentManager.beginTransaction().replace(android.R.id.content, frag).commit()
+
+            if (frag.arguments == null) {
+                val rootView = findViewById(android.R.id.content)
+                val viewTreeObserver = rootView.viewTreeObserver
+                if (viewTreeObserver.isAlive) {
+                    rootView.visibility = View.INVISIBLE
+                    viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
+                        override fun onGlobalLayout() {
+                            circularReveal()
+                            rootView.viewTreeObserver.removeOnGlobalLayoutListener(this)
+                        }
+                    })
+                }
+            }
         }
-        supportActionBar?.setDisplayShowHomeEnabled(true)
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun onBackPressed() {
```

Comment:
I haven't checked, but does this work if you click the back icon at the top left? I think it probably does because that will default to a onBackPressed(), but not sure